packages/shared/src/domain/contact/phone.ts

```diff
@@ -0,0 +1,44 @@
+import { stripNonNumericChars } from "../../common/string";
+
+export const phoneLength = 10;
+export const examplePhoneNumber = "1231231234";
+
+export function isPhoneValid(phone: string): boolean {
+  if (!phone) return false;
+  if (phone.length !== phoneLength) return false;
+  if (phone.match(/\D/)) return false; // needed as parseInt ignores non-numeric suffix
+  const phoneNumber = parseInt(phone, 10);
+  return !isNaN(phoneNumber);
+}
+
+/**
+ * Normalize a telephone number to a 10-digit string. Removes all non-numeric characters.
+ * This should not be used in code that parses phone numbers from user input, as it will
+ * remove the leftmost digits if the phone number is longer than 10 digits, which can lead to
+ * phone numbers with extensions having the extensions and missing important parts of the actual
+ * phone number.
+ * @param telephone
+ */
+export function normalizePhoneNumber(telephone: string): string;
+/**
+ * Normalize a telephone number to a 10-digit string. Removes all non-numeric characters.
+ * If in strict mode, only normalizes 10 to 11 digits strings (don't transform incorrect values
+ * with extensions and etc.).
+ * This should not be used with strict=false in code that parses phone numbers from user input, as
+ * it will remove the leftmost digits if the phone number is longer than 10 digits, which can lead
+ * to phone numbers with extensions having the extensions and missing important parts of the actual
+ * phone number.
+ * @param telephone the phone number to be normalized
+ * @param strict whether to normalize only 10 to 11 digits strings (default: false)
+ */
+export function normalizePhoneNumber(telephone: string, strict: boolean): string;
+export function normalizePhoneNumber(telephone: string, strict = false): string {
```

Comment:
should we set strict to `true` by default?

AFAIK we'll want to run this as strict in more places than the other way around

Reply:
Yeah, I think that's fair. I was trying to keep consistency with existing behavior - that didn't account for this, but maybe that was just a hidden bug for a situation we didn't expect at first.

I was thinking that we would want diff behavior for cx input than HIE's input, but I think it should be the same for both - with an update:
- if length <= 11 digits: use the 10 rightmost digits
- otherwise: use the 10 leftmost digits

Thoughts?

Reply:
Outdated, now we have an autofix instead of strict, and the logic changed considerably